Tyk Charts 1.4. release notes (#4495)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-charts/release-notes/version-1.4.md

---
title: Tyk Charts 1.4 Release Notes
date: 2024-02-05T15:49:11Z
description: "Release notes documenting updates, enhancements and changes for Tyk Charts versions within the 1.4 series."
tags: ["Tyk Charts", "Release notes", "changelog", "v1.4" ]
---

****Open Source** ([Mozilla Public License](https://github.com/TykTechnologies/tyk/blob/master/LICENSE.md))**

**This page contains all release notes for version 1.4.X displayed in a reverse chronological order**

### Support Lifetime
<!-- Required. replace X.Y with this release and set the correct quarter of the year -->
Our minor releases are supported until our next minor comes out. 

---

## 1.4.0 Release Notes

> ##### Release Date -- XX Apr 2024

Reviewer: Remember to update release date and to tag the release for Tyk Charts in GHub repository when released

Additionally, a disclaimer statement was added below the table, for customers to check that the third-party dependency they decide to install remains in support.

An example is given below for illustrative purposes only. Tested Versions and Compatible Versions information will require discussion with relevant squads and QA. -->

| Third Party Dependency                                     | Tested Versions        | Compatible Versions    | Comments | 
| ---------------------------------------------------------- | ---------------------- | ---------------------- | -------- | 
| [Kubernetes](https://kubernetes.io)                        | 1.26.x, 1.27.x, 1.28.x, 1.29.x | 1.19+          |          | 
| [Helm](https://helm.sh)                                    | 3.14.x                 | 3.x                    |          | 
| [Redis](https://redis.io)                                  | 6.2.x, 7.x    | 6.2.x, 7.x    | Used by Tyk Gateway and Dashboard | 
| [MongoDB](https://www.mongodb.com/try/download/community)  | 5.0.x, 6.0.x, 7.0.x | 4.4.x, 5.0.x, 6.0.x, 7.0.x | Used by Tyk Dashboard | 

> | [PostgreSQL](https://www.postgresql.org/download/)         | 11.x - 15.x LTS        | 11.x - 15.x            | Used by Tyk Dashboard | 

Reviewer 2: aren't Mongo and postgres used by other components like pump?

Author: updated

Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations
<!-- Required. Use the following statement if there are no deprecations, or explain if there are -->

> - In `tyk-dashboard` chart, `dashboard.hashKeys` field is deprecated and be replaced with `.global.hashKeys`. This is to ensure Dashboard, Gateway, and MDCB always get the same hashKeys configurations. Setting `dashboard.hashKeys` will no longer take effect. Please only use `.global.hashKeys` field.

Reviewer: ```suggestion
- In the `tyk-dashboard` chart, the `dashboard.hashKeys` field is deprecated and has been replaced with `.global.hashKeys`. This is to ensure Dashboard, Gateway, and MDCB always get the same hashKeys configurations. Setting `dashboard.hashKeys` will no longer take effect. Please only use `.global.hashKeys` field.
```

<!-- Optional section!
Used to share and notify users about our plan to deprecate features, configs etc. 
Once you put an item in this section, we must keep this item listed in all the following releases till the deprecation happens
##### Future deprecations. -->

#### Upgrade instructions
<!-- Required. For patches release (Z>0) use this: -->
For users currently on v1.3.x, we strongly recommend promptly upgrading to the latest release. 
<br/>
<!-- Go to the [Upgrading Tyk](#upgrading-tyk) section for detailed upgrade Instructions.
-->
You can use helm upgrade to upgrade your release

```bash
helm repo add tyk-helm https://helm.tyk.io/public/helm/charts/
helm repo update

helm upgrade [RELEASE_NAME] tyk-helm/[CHART_NAME]
```

#### Release Highlights
<!-- Required. Use similar ToV to previous release notes. For example for a patch release: -->

> ##### GA release of tyk-control-plane chart and tyk-mdcb chart

Reviewer: By GA do you mean General Availability?

```suggestion
##### General availability release of tyk-control-plane chart and tyk-mdcb chart
```

Author: updated

Here it is important to explain the benefit of each changelog item. As mentioned by James in a previous Slack message (https://tyktech.slack.com/archives/C044R3ZTN6L/p1686812207060839?thread_ts=1686762128.651249&cid=C044R3ZTN6L):
"...it is important to document the customer impact for the work delivered, so we can share it with prospects/install base. For example:
"New Chart delivers x and y benefit to a and b customer use cases. The business impact for them will be this and that" -->

##### Added
<!-- This section should be a bullet point list of new features. Explain:

- The purpose of the new feature
- How does the new feature benefit users?
- Link to documentation of the new feature
- For OSS - Link to the corresponding issue if possible on GitHub to allow the users to see further info.

Each change log item should be expandable. The first line summarises the changelog entry. It should be then possible to expand this to reveal further details about the changelog item. This is achieved using HTML as shown in the example below. -->

<ul>

<li>
<details>

> <summary>OSS: Simplify Tyk Operator Setup with Kubernetes Secret Creation</summary>

Reviewer: make Setup lowercase and Creation lower case

Author: updated

  # It also only works when TYK_MDCB_FORWARDANALYTICSTOPUMP is set to true.
  enableMultipleAnalyticsKey: true

  # This should be set to true if you choose not to store selective analytics
  dontStoreSelective: false

  # This should be set to true if you choose not to store aggregate analytics
  dontStoreAggregate: false

  # If set to true then it will not store analytics for tags having prefix specified in the list.
  # NB: Prefix “key-” is added in the list by default. This tag is added by gateway for keys.
  ignoreTagPrefixList: []

  # If enabled, it will store analytics for all the endpoints, irrespective of Track Endpoint plugin.
  trackAllPaths: false

  # If enabled, aggregate data will be generated per minute.
  storeAnalyticsPerMinute: false   
```
</details>
</li>

<li>
<details>
<summary>Tyk Control Plane: Added option to enable Dashboard hybrid organisation</summary>
We've added a convenient option to enable dashboard hybrid organization during bootstrapping. This eliminates the manual step of calling the Dashboard Admin API post-deployment to enable hybrid organization, which is essential for MDCB deployment.
</details>
</li>

<li>
<details>

> <summary>Gateway/Pump: Enhanced Pod security with `runAsNonRoot`</summary>

Reviewer: Make Pod lowercase in summary headings here and below

Author: @dcs3spp Pod is a Kubernetes special term (like Secret), maybe we should keep them as upper case?

> To harden security, we have added `runAsNonRoot: true` in the pod's securityContext. This prevents the Pods from running as root users, ensuring compatibility with the [*Restricted* Pod Security Policy](https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted).

Reviewer 2: ```suggestion
To harden security, we have added `runAsNonRoot: true` in the pod's `securityContext`. This prevents the Pods from running as root users, ensuring compatibility with the [*Restricted* Pod Security Policy](https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted).
```

Author: updated

</details>
</li>

<li>
<details>
<summary>Gateway: Allow Gateway to be updated if secret value is updated</summary>
We've introduced an annotation with a checksum of the secret as a value, triggering a deployment change when the secret is updated. This ensures that pods are replaced promptly, immediately utilizing the new values from the secret. This logic applies if `global.secrets.useSecretName` is not set, as the secret is then not part of the chart.
</details>
</li>

<li>
<details>
<summary>Customizable Pod Labels Across All Components</summary>

> Now, you have the flexibility to customize podLabels in all component charts. Simply populate the `podLabels` field with your desired content, and it will be added as pod labels.

Reviewer 2: What do you think? i might be wrong
```suggestion
Now, you have the flexibility to customize pod labels in all component charts. Simply populate the `podLabels` field with your desired content, and it will be added as pod labels.
```

Reviewer: Good spot @buraksekili 👍 I think podLabels is a field in a chart so the first sentence might be a typo? @caroltyk are you ok with the suggestion made by @buraksekili?

Author: changes made

</details>
</li>

<li>
<details>
<summary>Portal: Customizable Pod annotations in tyk-dev-portal</summary>

> We've added a `podAnnotations` field to the `tyk-dev-portal` chart, allowing you to customize pod annotations. Fill in the podAnnotations field with your specific content, and it will be added as pod annotations.

Reviewer 2: ```suggestion
We've added a `podAnnotations` field to the `tyk-dev-portal` chart, allowing you to customize pod annotations. Fill in the `podAnnotations` field with your specific content, and it will be added as pod annotations.
```

<ul>

<li>
<details>
<summary>Gateway/Pump: Removed the command in Gateway and Pump pod templates</summary>
We've removed unnecessary commands from the Gateway and Pump pod templates, allowing for the utilization of entrypoint scripts.
</details>
</li>

<li>
<details>
<summary>Dashboard: Allow arbitary image tags in tyk-dashboard</summary>
Now, you can use arbitrary image tags, including non-Semantic Versioning tags like `latest` for Dashboard. We've bypassed version checking in the Dashboard Deployment template to accommodate this flexibility.
</details>
</li>

<li>
<details>
<summary>Dashboard: Classic portal bootstrapping disabled by default</summary>
To avoid confusion with the latest Developer Portal, Classic Portal bootstrapping is now disabled by default in the Dashboard. If you wish to utilize the Classic Portal, simply enable it by setting `tyk-bootstrap.bootstrap.portal` to `true` in either the Tyk Stack or Tyk Control Plane chart.
</details>
</li>

<li>
<details>
<summary>Dashboard: Deprecation of `hashKeys` field</summary>

> The `dashboard.hashKeys` field is now deprecated. Instead, users should utilize the `global.hashKeys` field to set key hashing. This ensures configuration alignment across gateway, dashboard, and MDCB components.

Reviewer: Capitalise gateway and dashboard for consistency here with other usage in this release note

Author: updated

</details>
</li>

</ul>

##### Fixed

<ul>
<li>
<details>
<summary>Global: Redis TLS version specification</summary>

> We've corrected a typo in the values.yaml file within the "global.redis" section. The fields `sslMinVersion` and `sslMaxVersion` have been updated to `tlsMinVersion` and `tlsMaxVersion`, respectively. This ensures accurate specification of the Redis TLS version for enhanced security.

Reviewer 2: I could not get this log. Please correct me if I am wrong but did we have `sslMinVersion` and `sslMaxVersion` on values.yaml file at all? I mean, we've added `tlsMinVersion` and `tlsMaxVersion` to values.yaml file; however, the change only affected underlying environment variables. If I am not wrong, `tlsMinversion` and `tlsMaxVersion` were only added to support the new Redis configuration on Tyk - not to fix typo in values.yaml file as stated in the changelog

Author: @buraksekili this PR: https://github.com/TykTechnologies/tyk-charts/pull/267/files

Reviewer 2: thanks!!

---
title: Tyk Charts 1.4 Release Notes
date: 2024-02-05T15:49:11Z
description: "Release notes documenting updates, enhancements and changes for Tyk Charts versions within the 1.4 series."
tags: ["Tyk Charts", "Release notes", "changelog", "v1.4" ]
---

****Open Source** ([Mozilla Public License](https://github.com/TykTechnologies/tyk/blob/master/LICENSE.md))**

**This page contains all release notes for version 1.4.X displayed in a reverse chronological order**

### Support Lifetime
<!-- Required. replace X.Y with this release and set the correct quarter of the year -->
Our minor releases are supported until our next minor comes out. 

---

## 1.4.0 Release Notes

##### Release Date -- Apr 2024

#### Breaking Changes
<!-- Required. Use the following statement if there are no breaking changes, or explain if there are -->

> For MongoDB users: Tyk Charts 1.4.0 uses `mongo-go` as the default driver to connect to MongoDB. `mongo-go` driver is compatible with MongoDB 4.4.x and above. For MongoDB versions prior to 4.4, please set `global.mongo.driver` to `mgo`. We recommend reading [Choose a MongoDB driver]({{<ref "/planning-for-production/database-settings/mongodb#choose-a-mongodb-driver">}}) when you need to change driver setting.

Reviewer: ```suggestion
For MongoDB users: Tyk Charts 1.4.0 uses `mongo-go` as the default driver to connect to MongoDB. `mongo-go` driver is compatible with MongoDB 4.4.x and above. For MongoDB versions prior to 4.4, please set `global.mongo.driver` to `mgo`. We recommend reading [choose a MongoDB driver]({{<ref "/planning-for-production/database-settings/mongodb#choose-a-mongodb-driver">}}) when you need to change driver setting.
```

| [PostgreSQL](https://www.postgresql.org/download/)         | 11.x - 15.x LTS        | 11.x - 15.x            | Used by Tyk Dashboard | 

Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations
<!-- Required. Use the following statement if there are no deprecations, or explain if there are -->
There are no deprecations in this release.

<!-- Optional section!
Used to share and notify users about our plan to deprecate features, configs etc. 
Once you put an item in this section, we must keep this item listed in all the following releases till the deprecation happens. -->
##### Future deprecations
- In `tyk-dashboard` chart, `dashboard.hashKeys` field will be deprecated in future and be replaced with `.global.hashKeys`. This is to ensure Dashboard, Gateway, and MDCB always get the same hashKeys configurations. It is recommended users do not set `dashboard.hashKeys` and only use `.global.hashKeys` field.

#### Upgrade instructions
<!-- Required. For patches release (Z>0) use this: -->

> For users currently on v1.2.x, we strongly recommend promptly upgrading to the latest release. 

Reviewer: What about upgrade instructions from Tyk Charts 1.3 also?

<br/>
<!-- Go to the [Upgrading Tyk](#upgrading-tyk) section for detailed upgrade Instructions.
-->
You can use helm upgrade to upgrade your release

```bash
helm repo add tyk-helm https://helm.tyk.io/public/helm/charts/
helm repo update

helm upgrade [RELEASE_NAME] tyk-helm/[CHART_NAME]
```

> #### Release Highlights

Reviewer: should Tyk Charts 1.3 be replace with Tyk Charts 1.4 in this section?

The control plane should be deployed with same `global.mdcbSynchronizer.enabled` value too.
</details>
</li>

<li>
<details>
<summary>Gateway: Customise ServiceAccount to be used</summary>

Allow users to customise `serviceAccountName` for gateway, the name of the [Service Account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) that is going to be used by the Pods.
</details>
</li>

<li>
<details>
<summary>Gateway: Make service port name configurable</summary>

Users can configure Tyk Gateway service port name and Tyk Gateway control service port name. Default is `http`.
</details>
</li>

<li>
<details>
<summary>Gateway: Make initContainer image configurable</summary>

Users can configure Tyk Gateway initContainer image so that it is possible to load busybox image from preferred registry.
</details>
</li>

<li>
<details>
<summary>Dashboard: Added option to enable Tyk Identity Broker (TIB) in Tyk Dashboard</summary>

You can enable Internal [Tyk Identity Broker (TIB)]({{<ref "tyk-identity-broker">}}) in Tyk Dashboard by field `tyk-dashboard.tib.enabled` to `true`.
</details>
</li>

<li>
<details>

> <summary>Dashboard: Customise ServiceAccount to be used</summary>

Reviewer: ```suggestion
<summary>Dashboard: Customise ServiceAccount</summary>
```

Allow users to customise `serviceAccountName` for dashboard, the name of the [Service Account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) that is going to be used by the Pods.
</details>
</li>

<li>
<details>
<summary>Dashboard: Make service port name configurable</summary>

> Users can configure Tyk Dashboard service port name. Default is `http`.

Reviewer: What are the valid values, http, https etc? Should the changelog item be renamed to make it clearer that relates to the service protocol or does this relate to something else?

</details>
</li>

<li>
<details>
<summary>Pump: Added Graph pump support</summary>

[Graph Pumps]({{<ref "tyk-stack/tyk-pump/tyk-pump-configuration/graph-pump">}}) will be added when the user adds `mongo` or `postgres` to `pump.backend`. When `mongo` is added to `pump.backend` the Graph MongoDB Pump will be enabled. When `postgres` is added to `pump.backend` the Graph SQL Pump and Graph SQL Aggregate Pump will be enabled.
</details>
</li>

<li>
<details>
<summary>Pump: Customise ServiceAccount to be used</summary>

Allow users to customise `serviceAccountName` for pump, the name of the [Service Account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) that is going to be used by the Pods.
</details>
</li>

<li>
<details>
<summary>Pump: Make service port name configurable</summary>

> Users can configure Tyk Pump service port name. Default is `http`.

Reviewer: Same comment as for Dashboard service port name above and applicable to items below where service port name is configurable.

</details>
</li>

<li>
<details>

> <summary>Portal: Customise ServiceAccount to be used</summary>

Reviewer: ```suggestion
<summary>Portal: Customise ServiceAccount</summary>
```

</details>
</li>

<li>
<details>
<summary>New umbrella chart to deploy Tyk Control Plane</summary>

A new [Tyk Control Plane umbrella chart](https://github.com/TykTechnologies/tyk-charts/tree/main/tyk-control-plane) has been added to deploy Tyk Control Plane. 
It is currently in Beta. For installation instructions and configurations, please
read [Tyk Control Plane chart]({{<ref "product-stack/tyk-charts/tyk-control-plane-chart">}}).
</details>
</li>

</ul>

  
##### Changed
<!-- This should be a bullet-point list of updated features. Explain:

- Why was the update necessary?
- How does the update benefit users?
- Link to documentation of the updated feature
- For OSS - Link to the corresponding issue if possible on GitHub to allow the users to see further info.

Each change log item should be expandable. The first line summarises the changelog entry. It should be then possible to expand this to reveal further details about the changelog item. This is achieved using HTML as shown in the example below. -->
<ul>
<li>
<details>
<summary>Global config: Update default MongoDB driver to `mongo-go`</summary>

> Tyk Charts 1.4.0 uses `mongo-go` as the default driver to connect to MongoDB. `mongo-go` driver is compatible with MongoDB 4.4.x and above. For MongoDB versions prior to 4.4, please change `global.mongo.driver` to `mgo`. We recommend reading [Choose a MongoDB driver]({{<ref "planning-for-production/database-settings/mongodb#choose-a-mongodb-driver">}}) when you need to change driver setting.

Reviewer: ```suggestion
Tyk Charts 1.4.0 uses `mongo-go` as the default driver to connect to MongoDB. `mongo-go` driver is compatible with MongoDB 4.4.x and above. For MongoDB versions prior to 4.4, please change `global.mongo.driver` to `mgo`. We recommend reading [choose a MongoDB driver]({{<ref "planning-for-production/database-settings/mongodb#choose-a-mongodb-driver">}}) when you need to change driver setting.
```
